libs/ui/src/screens/settings/screens/lock-time-selector/lock-time-selector.tsx

```diff
@@ -1,17 +1,40 @@
 import React, { FC } from 'react';
+import { useDispatch } from 'react-redux';
 
 import { Dropdown } from '../../../../components/dropdown/dropdown';
+import { Option } from '../../../../components/dropdown/option.interface';
 import { useNavigation } from '../../../../hooks/use-navigation.hook';
+import { BackgroundMessager } from '../../../../messagers/background-messager';
 import { ModalContainer } from '../../../../modals/components/modal-container/modal-container';
+import { setAppLockTimePeriod } from '../../../../store/settings/settings.actions';
+import { useLockTimePeriodSelector } from '../../../../store/settings/settings.selectors';
+import { isWeb } from '../../../../utils/platform.utils';
 
 import { lockTimes } from './constants';
 
 export const LockTimeSelector: FC = () => {
   const { goBack } = useNavigation();
+  const dispatch = useDispatch();
+  const lockTimePeriod = useLockTimePeriodSelector();
+
+  const onSelect = ({ value }: Option<number>) => {
+    if (isWeb) {
+      BackgroundMessager.setLockTimePeriod(value);
+      dispatch(setAppLockTimePeriod(value));
```

Comment:
I don't like that we have 2 sources of through here.  
Ideally is to use a single value from `browser.storage.session` in UI and Background scripts